androidApp/src/main/java/com/example/tasky/android/common/viewmodel/MainViewModel.kt

```diff
@@ -4,22 +4,33 @@ import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import com.example.tasky.agenda.domain.IAgendaRepository
 import com.example.tasky.login.domain.manager.LoginManager
+import dev.tmapps.konnection.Konnection
 import kotlinx.coroutines.flow.SharingStarted
-import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.flow.combine
 import kotlinx.coroutines.flow.stateIn
 import kotlinx.coroutines.launch
 
 class MainViewModel(
     loginManager: LoginManager,
+    konnection: Konnection,
     private val agendaRepository: IAgendaRepository,
 ) : ViewModel() {
     val isLoginFlow =
-        loginManager.isLoginFlow
-            .onEach { isLoggedIn ->
-                if (isLoggedIn) {
-                    viewModelScope.launch {
-                        agendaRepository.syncAgenda()
+        loginManager.isLoginFlow.stateIn(scope = viewModelScope, started = SharingStarted.Eagerly, initialValue = false)
+
+    init {
+        viewModelScope.launch {
+            konnection
+                .observeHasConnection()
+                .combine(loginManager.isLoginFlow) { hasConnection, isLogin ->
+                    isLogin && hasConnection
+                }.collect {
+                    if (it) {
+                        viewModelScope.launch {
+                            agendaRepository.syncAgenda()
+                        }
```

Comment:
A shorthand for `viewModelScope. launch { flow. collect() }` is to use the `launchIn()` function. This keeps your code less nested.

```
combine(
  konnection.observeHasConnection(),
  loginManager.isLoginFlow()
) { hasConnection, isLogin ->
  hasConnection & isLogin
}
.filter { it }
.onEach {
  viewModelScope.launch {
    agendaRepository.syncAgenda()
  }
}
.launchIn(viewModelScope)
```

I’ve also replaced the `if` check with a `filter` operation to further reduce the nesting.